nodebuilder/header/config.go

```diff
@@ -93,7 +93,7 @@ func (cfg *Config) Validate() error {
 		return fmt.Errorf("module/header: misconfiguration of p2p exchange server: %w", err)
 	}
 
-	if cfg.Client != nil {
+	if !cfg.Client.Empty() {
```

Comment:
This check is not necessary, subsequently the Empty method as well

Reply:
It is necessary bc bridge nodes will have "invalid" client configs

Reply:
Ahhh, we should have passed node type to Validate, so that we visible check for the Bridge, or at least left a comment

Reply:
checked for Bridge. Left a comment